systems/src/apu/mod.rs

```diff
@@ -0,0 +1,1396 @@
+        fn when_apu_bleed_valve_closed_on_shutdown_cooldown_period_is_skipped_and_apu_stops() {
+            let mut tester = tester_with().running_apu_without_bleed_air();
+
+            assert!(tester.apu_is_available());
+
+            // Move from Running to Shutdown turbine state.
+            tester = tester.master_off().run(Duration::from_millis(1));
+            // APU N reduces below 95%.
+            tester = tester.run(Duration::from_secs(5));
+
+            assert!(!tester.apu_is_available());
+        }
+
+        #[test]
+        fn when_master_sw_off_then_back_on_during_cooldown_period_apu_continues_running() {
+            let tester = tester_with()
+                .running_apu_with_bleed_air()
+                .and()
+                .master_off()
+                .run(Duration::from_millis(
+                    ElectronicControlBox::BLEED_AIR_COOLDOWN_DURATION_MILLIS,
+                ));
+
+            let tester = tester
+                .then_continue_with()
+                .master_on()
+                .run(Duration::from_millis(1));
+
+            assert!(tester.apu_is_available());
+        }
+
+        #[test]
+        #[timeout(500)]
+        fn when_apu_starting_and_master_plus_start_sw_off_then_apu_continues_starting_and_shuts_down_after_start(
+        ) {
+            let mut tester = tester_with()
+                .starting_apu()
+                .run(Duration::from_secs(APPROXIMATE_STARTUP_TIME / 2));
+
+            assert!(tester.get_n().get::<percent>() > 0.);
+
+            tester = tester
+                .then_continue_with()
+                .master_off()
+                .and()
+                .start_off()
+                .run(Duration::from_secs(APPROXIMATE_STARTUP_TIME / 2));
+
+            assert!(tester.get_n().get::<percent>() > 90.);
+
+            loop {
+                tester = tester.then_continue_with().run(Duration::from_secs(1));
+
+                if tester.get_n().get::<percent>() == 0. {
+                    break;
+                }
+            }
+        }
+
+        #[test]
+        fn when_apu_shutting_down_at_7_percent_n_air_inlet_flap_closes() {
+            let mut tester = tester_with().running_apu().and().master_off();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+
+                if tester.get_n().get::<percent>() <= 7. {
+                    break;
+                }
+            }
+
+            assert!(!tester.is_air_intake_flap_fully_open());
+        }
+
+        #[test]
+        #[timeout(500)]
+        fn apu_cools_down_to_ambient_temperature_after_running() {
+            let ambient = ThermodynamicTemperature::new::<degree_celsius>(10.);
+            let mut tester = tester_with()
+                .running_apu()
+                .ambient_temperature(ambient)
+                .and()
+                .master_off();
+
+            while tester.get_egt() != ambient {
+                tester = tester.run(Duration::from_secs(1));
```

Comment:
Can we want to assert how fast we reach ambient temperature? Do we have any data on that?

Reply:
This will be part of a separate PR once Komp provides data on it, which will be somewhere in February.